include/secp256k1.h

```diff
@@ -74,6 +74,37 @@ void secp256k1_context_destroy(
   secp256k1_context_t* ctx
 ) SECP256K1_ARG_NONNULL(1);
 
+/** Set a callback function to be called when an illegal argument is passed to
+ *  an API call. The philosophy is that these shouldn't be dealt with through a
+ *  specific return value, as calling code should not have branches to deal with
+ *  the case that this code itself is broken.
+ *  On the other hand, during debug stage, one would want to be informed about
+ *  such mistakes, and the default (crashing) may be inadvisable.
+ *  When this callback is triggered, the API function called is guaranteed not
+ *  to cause a crash, though its return value and output arguments are
+ *  undefined.
+ */
+void secp256k1_context_set_illegal_callback(
```

Comment:
I think you should comment that these two functions are not threadsafe, since everything else in the library is (assuming you lock your contexts appropriately).

Alternately, we could gain threadsafety by storing these in the secp256k1 contexts. I think this is a bad idea since it'd clutter up the API and almost nobody will be using these functions, just throwing it out there.